### PR TITLE
Use PHPUnit 9.6 to run Symfony's test suite

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | N/A
| License       | MIT
| Doc PR        | N/A

Let's use PHPUnit 9.6 to run test test suite if we can.

### CODE REVIEW

File: phpunit [hunk 1]
```diff
@@ -10,12 +10,10 @@ if (!file_exists(__DIR__.'/vendor/symfony/phpunit-bridge/bin/simple-phpunit')) {
     exit(1);
 }
 if (!getenv('SYMFONY_PHPUNIT_VERSION')) {
-    if (\PHP_VERSION_ID < 70200) {
-        putenv('SYMFONY_PHPUNIT_VERSION=7.5');
```
Comment (derrabus) [inline phpunit:14]: This was a leftover from Symfony 4 wasn't it?

**Comment by derrabus** [comment]
All right, the output gives us a couple of tasks to work on for PHPUnit 10.

cc @OskarStark

**Comment by xabbuh** [comment]
👍 I would have expected a lot of messages about the deprecated `setMethods()` method though (see #49218). 🤔

**Comment by OskarStark** [comment]
It was fixed in
* https://github.com/symfony/symfony/pull/49220

Referenced PR in https://github.com/symfony/symfony/issues/49218

**Comment by xabbuh** [comment]
Yeah, for that particular case which is a file that can be used by users of Symfony. But we make heavy use of `setMethods()` in our own tests.

**Comment by OskarStark** [comment]
CleanShot 2023-02-04 at 10 42 51@2x

Ok... but why 🤷‍♂️ ...

That will be a lot of work for the abstract test cases in `/Test/`

**Comment by OskarStark** [comment]
> That will be a lot of work for the abstract test cases in /Test/

Oh no, we already named them `*TestCase` 💪

**Comment by nicolas-grekas** [comment]
Can you please rebase ?

**Comment by derrabus** [comment]
Done.

**Comment by nicolas-grekas** [comment]
Cool, we have very few failures. Up to fix them?

**Comment by derrabus** [comment]
What's the fix for…

> Expecting E_ERROR and E_USER_ERROR is deprecated and will no longer be possible in PHPUnit 10.

? Registering a custom error handler? 🤔

**Comment by xabbuh** [comment]
I think it's easier: #49375 (I need to check all places though)

**Comment by derrabus** [comment]
🟢

File: phpunit (hunk 1 above)
Comment (nicolas-grekas) [reply phpunit:14]: I'd keep this, the bridge is still declared as compatible with <7.2

File: src/Symfony/Bridge/PhpUnit/Tests/DeprecationErrorHandler/ConfigurationTest.php [hunk 2]
```diff
@@ -482,10 +482,24 @@ public function testBaselineFileWriteError()
...
         chmod($filename, 0444);
-        $this->expectError();
-        $this->expectErrorMessageMatches('/[Ff]ailed to open stream: Permission denied/');
         $configuration = Configuration::fromUrlEncodedString('generateBaseline=true&baselineFile='.urlencode($filename));
-        $configuration->writeBaseline();
+
+        $this->expectException(\ErrorException::class);
+        $this->expectExceptionMessageMatches('/[Ff]ailed to open stream: Permission denied/');
+
+        set_error_handler(static function(int $errno, string $errstr, string $errfile = null, int $errline = null) {
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/PhpUnit/Tests/DeprecationErrorHandler/ConfigurationTest.php]: missing space after function :)

Comment (derrabus) [reply src/Symfony/Bridge/PhpUnit/Tests/DeprecationErrorHandler/ConfigurationTest.php]: 😓

**Comment by nicolas-grekas** [comment]
Thank you @derrabus.

**Pull request merged**